### PR TITLE
BK-3731: do not retry on read-timeouts

### PR DESCRIPTION
**Pull request by shimshon-stratoscale**
@Stratoscale/mgmt

### CODE REVIEW

File: hammock/templates/client/file.j2 [hunk 1]
```diff
@@ -76,7 +76,7 @@ class {{ class_name }}(object):
             # Configure the retry class with backoff_factor of 0.1, will result
             # in waiting of about 50s after 10 retries.
             # the wait function is {backoff factor} * (2 ^ ({retry index} - 1)
-            retries = retry.Retry(total=retries, backoff_factor=0.1)
+            retries = retry.Retry(total=retries, backoff_factor=0.1, read=0)
```
Comment (rom-stratoscale) [inline hammock/templates/client/file.j2]: where do u diffrentiatiate between GET and post/put?

Comment (shimshon-stratoscale) [inline hammock/templates/client/file.j2]: the retry.Retry object has a default setting:
Retry.DEFAULT_METHOD_WHITELIST = frozenset({'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT', 'TRACE'})

Comment (rom-stratoscale) [inline hammock/templates/client/file.j2]: ok, so you affect all of them. not only post/put. What am I missing?

**Comment by eyal-stratoscale** [comment]
ship it

**Comment by rom-stratoscale** [comment]
תסבירו לי... :(

File: hammock/templates/client/file.j2 (hunk 1 above)
Comment (ravid-stratoscale) [inline hammock/templates/client/file.j2]: Same as @rom-stratoscale asked, where are you changing the whitelist?

**Comment by ravid-stratoscale** [comment]
@shimshon-stratoscale what about the differentiation between the verbs?

**Pull request merged**